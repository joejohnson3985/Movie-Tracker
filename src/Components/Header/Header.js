import React, { Component} from 'react';
import { Link } from 'react-router-dom';
import Logo from '../../Images/movie-tracker-logo.png';
import './Header.scss'
import { NavLink } from 'react-router-dom'


class Header extends Component {
  
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className='header'>
        <Link to="/" className='logo'>
          <img src={Logo} alt='Movie Tracker Logo' />
        </Link>
        <nav>
          <NavLink exact={true} to="/" activeClassName='current-nav' className='nav-button'><h4>New Movies</h4></NavLink>
          <NavLink activeClassName='current-nav' to="/Login" className='nav-button'><h4>Log In</h4></NavLink>
          <NavLink activeClassName='current-nav' to="/SignUp" className='nav-button'><h4>Sign Up</h4></NavLink>
        </nav>
      </div>
    );
  }
}


export default Header;