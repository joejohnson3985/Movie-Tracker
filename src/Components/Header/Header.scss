@import '../../Utilities/_Variables.scss';

img {
  height: 150px;
}

.header {
  background: $primary-color;
}

.logo {
  display: flex;
  align-items: center;
}