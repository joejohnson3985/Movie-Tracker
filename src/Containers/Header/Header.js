import React, { Component } from "react";
import { Link, NavLink } from "react-router-dom";
import Logo from "../../Images/movie-tracker-logo.png";
import "./Header.scss";
import { connect } from "react-redux";
import { userLogin, isLoggedIn } from "../../Actions";

export class Header extends Component {
  handleLogout = () => {
    this.props.isLoggedIn(false);
    this.props.addCurrentUser({});
  };

  render() {
    let whatToRender;
    if (this.props.isLoggedIn) {
      whatToRender = (
        <nav>
          <NavLink
            exact={true}
            to="/"
            activeClassName="current-nav"
            className="nav-button"
            onClick={() => this.props.browseAll()}
          >
            <h4>Browse</h4>
          </NavLink>
          <NavLink
            activeClassName="current-nav"
            to="/Favorites"
            className="nav-button"
          >
            <h4>Favorites</h4>
          </NavLink>
          <NavLink
            activeClassName="current-nav"
            onClick={this.handleLogout}
            to="/Login"
            className="nav-button log-out"
          >
            <h4>Log Out</h4>
          </NavLink>
        </nav>
      );
    } else {
      whatToRender = (
        <nav>
          <NavLink
            exact={true}
            to="/"
            activeClassName="current-nav"
            className="nav-button"
            onClick={this.props.browseAll}
          >
            <h4>Browse</h4>
          </NavLink>
          <NavLink
            activeClassName="current-nav"
            to="/Login"
            className="nav-button"
          >
            <h4>Log In</h4>
          </NavLink>
          <NavLink
            activeClassName="current-nav"
            to="/SignUp"
            className="nav-button"
          >
            <h4>Sign Up</h4>
          </NavLink>
        </nav>
      );
    }
    return (
      <div className="header">
        <Link to="/" className="logo">
          <img src={Logo} alt="Movie Tracker Logo" />
        </Link>
        {whatToRender}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  isLoggedIn: state.isLoggedIn
});

export const mapDispatchToProps = dispatch => ({
  addCurrentUser: user => dispatch(userLogin(user))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
