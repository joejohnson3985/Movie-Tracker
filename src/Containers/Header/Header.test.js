import React, { Component } from "react";
import { Header, mapStateToProps, mapDispatchToProps } from "./Header.js";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

describe("Header", () => {
  let mockStore = {};
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Header addCurrentUser={mockFn} />);
  });
  it("should match snapshot with all correct data passed through", () => {
    wrapper = shallow(<Header isLoggedIn={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with all correct data passed through", () => {
    wrapper = shallow(<Header isLoggedIn={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip("should call handleLogout when log out button is clicked", () => {
    wrapper.instance().handleLogout();
    expect(wrapper.isLoggedIn).toHaveBeenCalledWith(false);
    expect(wrapper.addCurrentUser).toHaveBeenCalledWith({});
  });

  describe("mapStateToProps", () => {
    it("should return an object with movies, shows, user, and message", () => {
      let mockStateResult = { isLoggedIn: false };
      let mockState = {
        isLoggedIn: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockStateResult);
    });
  });

  describe("mapDispatchToProps", () => {
    it("should pass correct type when addCurrentUser is called", () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).addCurrentUser();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
