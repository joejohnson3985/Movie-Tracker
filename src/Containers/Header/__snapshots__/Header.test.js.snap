// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot with all correct data passed through 1`] = `ShallowWrapper {}`;

exports[`Header should match snapshot with all correct data passed through 2`] = `ShallowWrapper {}`;
